File: Domain/Project.cs
using System;
using System.Collections.Generic;

namespace Domain;

public class Project
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Code { get; set; } = string.Empty;
    public int? DepartmentId { get; set; }
    public Department? Department { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public ICollection<TimesheetEntry> TimesheetEntries { get; set; } = new List<TimesheetEntry>();
}

File: client/src/lib/types/project.ts
import type { Department } from './department';

export interface Project {
    id: number;
    name: string;
    code: string;
    description?: string;
    isActive: boolean;
    departmentId?: number | null;
    department?: Department | null;
}

export interface UpsertProjectRequest {
    id?: number;
    name: string;
    code?: string;
    description?: string;
    isActive?: boolean;
    departmentId?: number | null;
}
